### ISSUE TITLE
Run response-scorer training once real handwriting-label data exists (issue #81 follow-up)

### ISSUE THREAD

**Issue by hoj201** [hoj201/grader_bot#84]
## What

Actually run training for the response-scorer CRNN (issue #81) and export it
to `models/response_scorer/weights.onnx`, so the "CNN verifier
(experimental)" option in the Grade tab stops silently falling back to
Mathpix.

## Blocked on: real training data

Training was **deliberately postponed** rather than starting on synthetic-only
data (single-font renders) or MNIST (isolated digits only -- no `.`/`-`, no
multi-character sequences, no box-crop realism). Instead we built a way to
harvest trustworthy *real* handwriting samples first:

- **Handwriting Data tab** (`graderbot/handwriting_sample_worksheets.py` +
  `graderbot/handwriting_harvest.py`): generates an ordinary worksheet whose
  boxes already print their own answer, so students just copy it by hand.
  Because the printed text *is* the answer key, every scanned-back box
  becomes a verified `HANDWRITING_LABEL` row with zero manual review.

**Do not start this issue until a meaningful batch of real labels has been
collected** via that tab (print worksheets → have students copy → scan
back in → harvest). Check `storage.list_handwriting_labels(conn,
verified_only=True)` count, or the running total shown on the Handwriting
Data tab, before proceeding.

## Steps once there's data

1. Confirm there's enough real labeled data to be a meaningful validation
   set (rough starting point: a few hundred samples, spread across digits/
   decimals/negatives).
2. Extend `training/eval.py` to also evaluate against
   `list_handwriting_labels(verified_only=True)`, not just synthetic
   held-out data -- report the synthetic-vs-real accuracy gap (the number
   that would have caught the paused `pylaia-iam` spike's domain mismatch
   sooner, issue #73).
3. Run training on Modal (costs compute -- confirm with @hoj201 before
   kicking it off):
   ```shell
   poetry run modal run training/modal_app.py --steps 20000
   ```
   This writes `models/response_scorer/{weights.onnx,vocab.json}` straight
   into the repo.
4. Run `training/eval.py` (both synthetic and real-data eval) and report the
   numbers before deciding the Grade tab option is ready to lose its
   "(experimental)" label.
5. If real-eval accuracy lags synthetic, consider folding the real labels
   into fine-tuning (see issue #81's original phasing, Phase 7) once there's
   enough of them (rough threshold discussed: >150-200 labeled real crops).

## Related

Follows on from #81 (the design/scaffold, merged to `main`). Do not close
this until an actual trained model has been evaluated and shipped, not just
the training code existing.

🤖 _Posted by Claude Code on behalf of @hoj201_
